tools/ack-discover/ecrpublic.py

```diff
@@ -0,0 +1,136 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import dataclasses
+import uuid
+
+import boto3
+
+BASE_ECR_URL = "public.ecr.aws/aws-controllers-k8s"
+ECR_PUBLIC_AWS_ACCOUNT_ID = "628432846661"
```

Comment:
should we not have this accountID in source code but instead inject as env variable?

Reply:
I wouldn't need it at all if ECR Public allowed me to pass the alias `aws-controllers-k8s` for `registryId` :)

But it's not secret information, if that's what you're concerned about. It's the same as a normal ECR (non-ECR Public) repository including the AWS account ID in the image URIs... i.e. [{ACCOUNT_ID}:dkr.ecr.us-west-2.amazonaws.com...](https://github.com/aws/amazon-vpc-cni-k8s/blob/0dd549fdd759f5d4fda95ee26859a52b1fe0e075/config/master/aws-k8s-cni.yaml#L124)